### PR TITLE
[iOS][Android] Upgrade react-navigation to v6 

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-4149.
Closes ENG-4150.
Closes ENG-4160.

The react-navigation v5 doesn't work with the react-native 0.70. So, I've migrated NCL and Home to use v6.

# How

- Bumped versions of related packages.
- Went through the migration guide. 
- Disable react-native-screens integration on Android in Home.
 
> Note:
I'm aware of [proposed fix](https://github.com/expo/expo/pull/16333) which fixes integration between screens and navigation on Android. However, I'm not sure if that change is safe. That's why, for now, `react-native-screens` were disabled. 

# Test Plan

- expo-go - home ✅
- expo-go - NCL ✅
- bare-expo - NCL ✅

### CODE REVIEW

File: home/navigation/Navigation.types.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export type ModalStackRoutes = {
   QRCode: undefined;
+  RootStack: undefined;
```
Comment (tsapeta) [inline home/navigation/Navigation.types.ts:3]: This type looks weird, do we really need it? 😅

**Review by tsapeta — APPROVED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -3354,23 +3354,15 @@
...
 
-"@react-navigation/bottom-tabs@~5.11.1":
-  version "5.11.15"
-  resolved "https://registry.yarnpkg.com/@react-navigation/bottom-tabs/-/bottom-tabs-5.11.15.tgz#f973625cc32d9c5a4067851f084cb11ccd68fe79"
-  integrity sha512-TBY419W6aN/HZg98xbVp5Bx1HEF5sXuHR5f55W6KMI4k2AvxlwelKD1wbfvEcX2iuQT0YUiiXsACRFUSECYhkw==
-  dependencies:
-    color "^3.1.3"
-    react-native-iphone-x-helper "^1.3.0"
-
 "@react-navigation/core@5.15.1":
```
Comment (Kudo) [inline yarn.lock:3366]: just noticed that we still have react-navigation v5 in expo-dev-menu and expo-dev-launcher. do we plan to upgrade them together?

Comment (FiberJW) [reply yarn.lock:3366]: I agree with the concern about upgrading these packages. When I tried upgrading to v6, it required me to do so for every package, or else I'd have problems with the monorepo's resolution of react-navigation.

Comment (lukmccall) [reply yarn.lock:3366]: I will upgrade that, but later. Right now, it's frozen anyway, because we're not rebuilding dev bundles.

Comment (Kudo) [reply yarn.lock:3366]: when i reverted [previous react-navigation upgrade change](https://github.com/expo/expo/pull/18088), it was focusing on the inconsistent version between bare-expo and home/ncl, so dev-client should not be a problem. i'm re-running the failed test-suites and makes sure it's passed.

Comment (lukmccall) [reply yarn.lock:3366]: `Android Instrumentation Tests` failed, but it's not related. I'll merge that PR

**Review by Kudo — APPROVED** [review]

**Pull request merged**